pkg/clients/clients.go

```diff
@@ -196,6 +198,33 @@ func (k *k8sClients) GetPlatform() (string, error) {
 	}
 }
 
+func (k *k8sClients) GetNodesByLabels(ctx context.Context, matchingLabels map[string]string) (*v1.NodeList, error) {
+	nodes, err := k.clientset.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: labels.Set(matchingLabels).String()})
+	if err != nil {
+		return nil, err
+	}
+
+	// filter nodes by taints
+	nodesWithoutTaints := nodes.Items[:0]
```

Comment:
I'm curious about this. Does it have any performance gains or just a shortcut to e.g. make sure the types are the same?

Reply:
This is also not immediately obvious to me. I would probably use `make()` here instead:
```suggestion
	nodesWithoutTaints := make([]v1.Node, 0, len(nodes.Items))
```

Reply:
it defines a new slice over existing array, so no new allocation  is needed in this case

Reply:
Right. I think this indeed works as expected right now, but I think it might be error-prone if we decide to re-iterate over the original node slice for some reason. Could we unit-test this method?

Reply:
it is actually a more accepted way in GO ( saves you the allocation of the undelaying array

Reply:
I think altering a collection while iterating over it is always a risky business.
I could live with here it if we had unit tests for the method 😃

Reply:
i will add a unit-test. Just one notice: we are not altering a collection (it is risky in a loop without indices), we are creating a new slice and adding the data from the underlying array to it ( not an issue)

Reply:
fixed